cmd/modelcmd/modelcommand.go

```diff
@@ -158,7 +183,10 @@ func (c *ModelCommandBase) initModel0() error {
 func (c *ModelCommandBase) SetModelName(modelName string, allowDefault bool) error {
 	c._modelName = modelName
 	c.allowDefaultModel = allowDefault
-	if c.runStarted {
+
+	// After setting the model name, we may need to ensure we have access to the
+	// other model details if not already done.
+	if c.initModelRequired() {
```

Comment:
This seems awkward - is there any way to move the checks that `initModelRequired` does inside `initModel`? I realise there are a couple of places that call `initModel` without being guarded. Or just having a method `maybeInitModel` instead of having to do the check and then call `initModel`.